src/core/utils/CorePlugin.ts

```diff
@@ -34,17 +35,29 @@ export class CorePlugin extends JWPlugin {
     // Public
     //--------------------------------------------------------------------------
 
-    enter(): void {
-        this.editor.vDocument.enter();
+    /**
+     * Handle the (shift+) enter key.
+     *
+     * @param intent
+     */
+    enter(intent: KeydownIntent): void {
+        if (intent.payload.shiftKey) {
+            this.editor.vDocument.insert(new VNode(VNodeType.LINE_BREAK));
+        } else {
+            this.editor.vDocument.enter();
+        }
     }
     /**
      * Insert something at range.
      *
      * @param intent
      */
     insert(intent: InsertIntent): void {
-        // TODO: check the intent to insert other things than text.
-        this.editor.vDocument.insertText(intent.payload.value);
+        if (typeof intent.payload.value === 'string') {
+            this.editor.vDocument.insertText(intent.payload.value);
+        } else {
+            this.editor.vDocument.insert(intent.payload.value);
+        }
```

Comment:
I would rename the current `insert` into `insertText` and then have `insert(node)` and `insertText(string)`